lib/vmpooler/pool_manager.rb

```diff
@@ -299,6 +299,47 @@ def destroy_vm(vm, pool)
       end
     end
 
+    def create_vm_disk(vm, disk_size)
+      Thread.new do
+        _create_vm_disk(vm, disk_size)
+      end
+    end
+
+    def _create_vm_disk(vm, disk_size)
+      host = $vsphere['disk_manager'].find_vm(vm) ||
+             $vsphere['disk_manager'].find_vm_heavy(vm)[vm]
+
+      if (host) && ((! disk_size.nil?) || (! disk_size.empty?))
+        $logger.log('s', "[ ] [disk_manager] '#{vm}' is attaching a #{disk_size}mb disk")
+
+        start = Time.now
+
+        template = $redis.hget('vmpooler__vm__' + vm, 'template')
+        datastore = nil
+
+        $config[:pools].each do |pool|
+          if pool['name'] == template
+            datastore = pool['datastore']
+          end
+        end
+
+        if ((! datastore.nil?) || (! datastore.empty?))
```

Comment:
This should be `&&`. It is possible for the `datastore` to not be nil (first check) by being empty.